src/context-parser.js

```diff
@@ -408,9 +406,17 @@ function Parser (config, listeners) {
         for (k in config) {
             self.config[k] = config[k];
         }
-    } else {
-        config = self.config;
     }
+    config = self.config;    
+
+    // config defaulted to false
+    config.enableInputPreProcessing = (config.enableInputPreProcessing === true);
+    config.enableCanonicalization = (config.enableCanonicalization === true);
+    config.enableVoidingIEConditionalComments = (config.enableVoidingIEConditionalComments === true);
```

Comment:
please create the corresponding PR to the html-purifier and secure-handlebars for renaming this conf param